packages/manager/apps/web/client/app/hosting/database/copy/hosting-database-copy.controller.js

```diff
@@ -0,0 +1,138 @@
+import {
+  COPY_TYPE_MAIN_DATABASE,
+  COPY_TYPE_WEBCLOUD_DATABASE,
+} from './hosting-database-copy.constants';
+
+angular.module('App').controller(
+  'HostingDatabaseCopyCtrl',
+  class HostingDatabaseCopyCtrl {
+    constructor(
+      $http,
+      $q,
+      $translate,
+      $scope,
+      $state,
+      $window,
+      Alerter,
+      coreURLBuilder,
+    ) {
+      this.$http = $http;
+      this.$q = $q;
+      this.$translate = $translate;
+      this.$scope = $scope;
+      this.$state = $state;
+      this.$window = $window;
+      this.Alerter = Alerter;
+      this.coreURLBuilder = coreURLBuilder;
+      this.COPY_TYPE_MAIN_DATABASE = COPY_TYPE_MAIN_DATABASE;
+      this.COPY_TYPE_WEBCLOUD_DATABASE = COPY_TYPE_WEBCLOUD_DATABASE;
+      this.selectedType = COPY_TYPE_MAIN_DATABASE;
+      this.selectedMain = null;
+      this.selectedDatabase = null;
+      this.selectedWebCloud = null;
+      this.selectedWebCloudDatabase = null;
+      this.targetWebCouldDatabases = null;
+      this.targetDatabases = null;
+      this.webCloudDatabases = null;
+    }
+
+    $onInit() {
+      this.$scope.copyDatabase = () => this.copyDatabase();
+      this.$scope.load = () => this.load();
+      this.$scope.isStep1Valid = () => this.isStep1Valid();
+      this.currentDatabase = this.$scope.currentActionData.currentDatabase;
```

Comment:
Question : What do you need to use `$scope` instead of controller itself ?
If it's really needed, it could be simplified as :
```suggestion
      this.$scope.copyDatabase = this.copyDatabase;
      this.$scope.load = this.load;
      this.$scope.isStep1Valid =  this.isStep1Valid;
      this.currentDatabase = this.$scope.currentActionData.currentDatabase;
```